src/validators/OidcRecoveryValidator.sol

```diff
@@ -17,48 +17,127 @@ import { Groth16Verifier } from "../autogenerated/JwtTxValidationVerifier.sol";
+  /// @param expectedValidator The expected address of the web authentication validator.
+  /// @param validator The address of the web authentication validator.
+  error InvalidWebAuthnValidatorAddress(address expectedValidator, address validator);
+
+  /// @notice Thrown when the transaction to address overflows.
+  error TransactionToOverflow();
+
+  /// @notice Thrown when trying to validate a non-function call transaction.
+  error OnlyFunctionCallsSupported();
+
+  /// @notice Thrown when trying to validate a transaction that does not call `addValidationKey` on the `WebAuthValidator` contract.
+  /// @param expectedSelector The expected selector of the `addValidationKey` function.
+  /// @param actualSelector The actual selector of the function being called.
+  error UnauthorizedFunctionCall(bytes4 expectedSelector, bytes4 actualSelector);
+
+  /// @notice Thrown when the passkey hash is invalid.
+  /// @param expectedHash The expected passkey hash.
+  /// @param actualHash The actual passkey hash.
+  error InvalidPasskeyHash(bytes32 expectedHash, bytes32 actualHash);
+
+  /// @notice Thrown when the account is not ready to recover.
+  error NotReadyToRecover();
+
+  /// @notice The data for an OIDC account.
+  /// @param oidcDigest The PoseidonHash(sub || aud || iss || salt) of the OIDC key.
+  /// @param iss The OIDC issuer.
+  /// @param readyToRecover True if the account is ready to recover, false otherwise.
+  /// @param pendingPasskeyHash The hash of the pending passkey.
+  /// @param recoverNonce The nonce for the recovery process to prevent replay attacks.
   struct OidcData {
-    bytes32 oidcDigest; // PoseidonHash(sub || aud || iss || salt)
+    bytes32 oidcDigest;
     string iss;
     bool readyToRecover;
     bytes32 pendingPasskeyHash;
     uint256 recoverNonce;
   }
 
+  /// @notice The data for an OIDC key creation.
+  /// @param oidcDigest The PoseidonHash(sub || aud || iss || salt) of the OIDC key.
+  /// @param iss The OIDC issuer.
   struct OidcCreationData {
-    bytes32 oidcDigest; // PoseidonHash(sub || aud || iss || salt)
+    bytes32 oidcDigest;
     string iss;
   }
 
+  /// @notice The data for a zk proof.
   struct ZkProof {
     uint[2] pA;
     uint[2][2] pB;
     uint[2] pC;
   }
 
+  /// @notice The data for starting a recovery process.
+  /// @param zkProof The zk proof.
+  /// @param issHash The hash of the OIDC issuer.
+  /// @param kid The key id (kid) of the OIDC key.
+  /// @param pendingPasskeyHash The hash of the pending passkey to be added.
   struct StartRecoveryData {
     ZkProof zkProof;
     bytes32 issHash;
     bytes32 kid;
     bytes32 pendingPasskeyHash;
   }
 
-  mapping(address => OidcData) accountData;
-  mapping(bytes32 => address) digestIndex;
+  /// @notice The mapping of account addresses to their OIDC data.
+  mapping(address account => OidcData oidcData) accountData;
 
+  /// @notice The mapping of OIDC digests to their corresponding account addresses.
```

Comment:
We should explan here that this is needed because we use this at recovery moment to get back the address of the user.